test/object-store/sync/flx_sync.cpp

```diff
@@ -4995,6 +4967,464 @@ TEST_CASE("flx: nested collections in mixed", "[sync][flx][baas]") {
+                    return std::nullopt;
+
+                using BatchState = sync::DownloadBatchState;
+                using Event = SyncClientHookEvent;
+                switch (data.event) {
+                    case Event::ErrorMessageReceived:
+                        REQUIRE(cur_state == TestState::start);
+                        REQUIRE(data.error_info);
+                        REQUIRE(data.error_info->raw_error_code == 200);
+                        REQUIRE(data.error_info->server_requests_action ==
+                                sync::ProtocolErrorInfo::Action::Transient);
+                        REQUIRE_FALSE(data.error_info->is_fatal);
+                        return TestState::reconnect_received;
+
+                    case Event::DownloadMessageReceived: {
+                        std::lock_guard lock(callback_mutex);
+                        // multi-message bootstrap in progress..
+                        ++download_msg_count;
+                        if (cur_state != TestState::reconnect_received && cur_state != TestState::downloading)
+                            return std::nullopt;
+
+                        if (data.batch_state == BatchState::MoreToCome) {
+                            // More than 1 bootstrap message, always a multi-message
+                            bootstrap_mode = BootstrapMode::MultiMessage;
+                            logger->trace("ROLE CHANGE: detected multi-message bootstrap");
+                            return TestState::downloading;
+                        }
+                        // single bootstrap message or last message in the multi-message bootstrap
+                        else if (data.batch_state == BatchState::LastInBatch) {
+                            if (download_msg_count == 1) {
+                                if (data.num_changesets == 1) {
+                                    logger->trace("ROLE CHANGE: detected single-message/single-changeset bootstrap");
+                                    bootstrap_mode = BootstrapMode::SingleMessage;
+                                }
+                                else {
+                                    logger->trace("ROLE CHANGE: detected single-message/multi-changeset bootstrap");
+                                    bootstrap_mode = BootstrapMode::SingleMessageMulti;
+                                }
+                            }
+                            return TestState::downloaded;
+                        }
+                        return std::nullopt;
+                    }
+
+                    // A bootstrap message was processed
+                    case Event::BootstrapMessageProcessed: {
+                        REQUIRE(data.batch_state != BatchState::SteadyState);
+                        REQUIRE((cur_state == TestState::downloading || cur_state == TestState::downloaded));
+                        if (data.batch_state != BatchState::SteadyState) {
```

Comment:
if we've required that batch_state != BatchState::SteadyState on like 5110, then do we need to check if it's not equal here?